app/app.go

```diff
@@ -384,135 +436,265 @@ func NewCanto(
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
 	)
 
-	// register the proposal types
-	govRouter := govtypes.NewRouter()
-	govRouter.AddRoute(govtypes.RouterKey, govtypes.ProposalHandler).
-		AddRoute(paramproposal.RouterKey, params.NewParamChangeProposalHandler(app.ParamsKeeper)).
-		AddRoute(distrtypes.RouterKey, distr.NewCommunityPoolSpendProposalHandler(app.DistrKeeper)).
-		AddRoute(upgradetypes.RouterKey, upgrade.NewSoftwareUpgradeProposalHandler(app.UpgradeKeeper)).
-		AddRoute(ibcclienttypes.RouterKey, ibcclient.NewClientProposalHandler(app.IBCKeeper.ClientKeeper)).
-		AddRoute(erc20types.RouterKey, erc20.NewErc20ProposalHandler(&app.Erc20Keeper)).
-		AddRoute(govshuttletypes.RouterKey, govshuttle.NewgovshuttleProposalHandler(&app.GovshuttleKeeper))
-
-	govKeeper := govkeeper.NewKeeper(
-		appCodec, keys[govtypes.StoreKey], app.GetSubspace(govtypes.ModuleName),
-		app.AccountKeeper, app.BankKeeper, &stakingKeeper, govRouter,
-	)
-
-	// Canto Keeper
 	app.InflationKeeper = inflationkeeper.NewKeeper(
-		keys[inflationtypes.StoreKey], appCodec, app.GetSubspace(inflationtypes.ModuleName),
-		app.AccountKeeper, app.BankKeeper, app.DistrKeeper, &stakingKeeper,
+		runtime.NewKVStoreService(keys[inflationtypes.StoreKey]),
+		appCodec,
+		app.GetSubspace(inflationtypes.ModuleName),
+		app.AccountKeeper,
+		app.BankKeeper,
+		app.DistrKeeper,
+		app.StakingKeeper,
 		authtypes.FeeCollectorName,
 	)
 
-	// register the staking hooks
-	// NOTE: stakingKeeper above is passed by reference, so that it will contain these hooks
-	// NOTE: Distr, Slashing and Claim must be created before calling the Hooks method to avoid returning a Keeper without its table generated
-	app.StakingKeeper = *stakingKeeper.SetHooks(
-		stakingtypes.NewMultiStakingHooks(
-			app.DistrKeeper.Hooks(),
-			app.SlashingKeeper.Hooks(),
-		),
-	)
-
 	app.Erc20Keeper = erc20keeper.NewKeeper(
-		keys[erc20types.StoreKey], appCodec, app.GetSubspace(erc20types.ModuleName),
-		app.AccountKeeper, app.BankKeeper, app.EvmKeeper,
+		runtime.NewKVStoreService(keys[erc20types.StoreKey]),
+		appCodec,
+		app.GetSubspace(erc20types.ModuleName),
+		app.AccountKeeper,
+		app.BankKeeper,
+		app.EvmKeeper,
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
 	)
 
 	app.GovshuttleKeeper = govshuttlekeeper.NewKeeper(
-		keys[govshuttletypes.StoreKey], appCodec, app.GetSubspace(govshuttletypes.ModuleName),
-		app.AccountKeeper, app.Erc20Keeper, govKeeper,
+		runtime.NewKVStoreService(keys[govshuttletypes.StoreKey]),
+		appCodec,
+		app.GetSubspace(govshuttletypes.ModuleName),
+		app.AccountKeeper,
+		app.Erc20Keeper,
+		govKeeper,
 	)
 
 	app.CSRKeeper = csrkeeper.NewKeeper(
-		appCodec, keys[csrtypes.StoreKey],
+		appCodec,
+		runtime.NewKVStoreService(keys[csrtypes.StoreKey]),
 		app.GetSubspace(csrtypes.ModuleName),
-		app.AccountKeeper, app.EvmKeeper, app.BankKeeper,
+		app.AccountKeeper,
+		app.EvmKeeper,
+		app.BankKeeper,
 		authtypes.FeeCollectorName,
 	)
```

Comment:
Some of Canto's custom modules which have Params, such as `inflation`, `govshuttle`, `csr`, and `onboarding`, lack implementation of `MsgUpdateParams` and its handler, or the authority for ParamChange Proposal for the gov module account isn't defined. It seems necessary to conduct comprehensive testing, verification, and code addition for these modules in relation to the above issue.

Reply:
added `MsgUpdateParams` and its implementation to csr, inflation, onborading modules

Reply:
In the case of the `govshuttle` module, the assignment of the `authority` to `NewKeeper()` does not seem to be fixed yet. Since these actions related to authority are major changes occurring in this PR, it would be good to add integration and unit test cases for regression testing so that they can be caught in the workflow when not operating

Reply:
The reason i didn't add `authority` and `MsgUpdateParams` to `govshuttle` is that the govshuttle module also has [Params](https://github.com/b-harvest/Canto/blob/dudong2/feat/canto-main-cosmos-sdk%40v0.50.x/proto/canto/govshuttle/v1/govshuttle.proto#L10), but they are empty. I skipped them because I don't need them right away.